pkg/types/interfacetests/chain_components_interface_tests.go

```diff
@@ -15,8 +15,28 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
 )
 
-type ChainComponentsInterfaceTester[T TestingT[T]] interface {
+type ChainComponentsTester[T any] interface {
```

Comment:
- defined `ChainComponentsTester` so we use the `Setup(t T, startCR bool)`  only for `chainComponents` tests
- defined `chainComponentsTestcase` so we use the `startCR` testcase flag only for `chainComponents` testcases
- defined `runChainComponentsTests` to execute `chainComponents` specific tests using the `ChainComponentsTester` interface with `Setup(t T, startCR bool)`